pkg/azureutils/azure_disk_utils.go

```diff
@@ -927,16 +935,22 @@ func UpdateCRIWithRetry(ctx context.Context, informerFactory azurediskInformers.
 
 		switch target := copyForUpdate.(type) {
 		case *diskv1beta1.AzVolume:
-			if cachedClient == nil {
+			if updateMode&UpdateCRIStatus > 0 && !reflect.DeepEqual(objForUpdate.(*diskv1beta1.AzVolume).Status, target.Status) {
```

Comment:
It's conventional to wrap the `&` expression in parentheses and use `!= 0` for flags checks, e.g. `(updateMode&UpdateCRIStatus) != 0`.